### ISSUE TITLE
Add Provenance Filter Functionality.

### ISSUE THREAD

**Issue by luciencd** [metpetdb/api_v2#4]
When you are logged in or out, the search functionality returns everything you have access to. However, if you only want to see public or private samples, and click the radio button for it, that does nothing to change the results.

Need to fix [query file](https://github.com/metpetdb/api_v2/blob/master/metpetdb_api/api/samples/lib/query.py) to accept provenance attribute (committed those changes already). Do this for chemical-analysis, and samples.

Next, I need to add a few tests to make sure that a user will access the correct information when they choose the provenance to be public or private.

**Comment by kristallizer** [comment]
@luciencd are those changes in this branch? https://github.com/metpetdb/api_v2/tree/provenance Can you open a pull request with that branch?

**Comment by luciencd** [comment]
Yeah. thats the changes that are in provenance. I was going to wait until getting the tests working for the feature to create the pull request, but I can do that now if you want.
